docs/source/coroutines.rst

```diff
@@ -260,6 +260,182 @@ And of course, the sky is the limit when you compose the two.
     This means that Tasks passed to it are *still running*.
     You may need to cancel those Tasks with :meth:`.Task.cancel`.
 
+.. _task_manager_tutorial:
+
+:class:`!TaskManager`: Structured Asymmetric Concurrency
+========================================================
+
+The :class:`~cocotb.triggers.TaskManager` class is another way to run multiple async routines concurrently and wait for them all to complete,
+but properly manages the lifetime of its "children" and handles exceptions and cancellations gracefully.
```

Comment:
```suggestion
The :class:`~cocotb.triggers.TaskManager` class is another way to run multiple async routines concurrently and wait for them all to complete.
It properly manages the lifetime of its "children" and handles exceptions and cancellations gracefully.
```
If the "but" was to contrast this part with the section above, maybe make it more explicit.